model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -2201,6 +2164,32 @@ async def execute(self, user: User, request: ModelDownloadRequest) -> ModelDownl
         return ModelDownloadResponse(urls=urls)
 
 
+def fill_hardware_info(request: CreateLLMModelEndpointV1Request):
```

Comment:
nit: should we prefix these module-level functions with an underscore? Don't want to tempt people to import those, when they should be going through the use case.